src/rust/libsql-shell/src/main.rs

```diff
@@ -38,6 +38,40 @@ fn execute(stmt: &mut Statement) -> Result<Vec<Vec<String>>> {
     Ok(rows.map(|r| r.unwrap()).collect())
 }
 
+struct StrStatements {
+    value: String,
+}
+
+impl Iterator for StrStatements {
+    type Item = String;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let mut embedded = false;
+        let mut pos = 0;
+        for (index, char) in self.value.trim().chars().enumerate() {
+            if char == '\'' {
+                embedded = !embedded;
+                continue;
+            }
+            if embedded || char != ';' {
+                continue;
+            }
+            let str_statement = self.value[pos..index+1].to_string();
+            if str_statement.starts_with(';') || str_statement.is_empty() {
+                pos = index+1;
+                continue;
+            }
```

Comment:
We could remove that here and filter iterator results but honestly idk which option is better.